govcd/vapp_test.go

```diff
@@ -541,7 +541,26 @@ func (vcd *TestVCD) Test_AddNewVMMultiNIC(check *C) {
 		fmt.Println("Skipping adding another vdc network as network2 was not specified")
 	}
 
-	task, err := vapp.AddNewVM(check.TestName(), vapptemplate, desiredNetConfig, true)
+	var task Task
+	var sp types.Reference
+	var customSP = false
+
+	if vcd.config.VCD.StorageProfile.SP1 != "" {
+		sp, _ = vcd.vdc.FindStorageProfileReference(vcd.config.VCD.StorageProfile.SP1)
+	}
+
+	if sp.HREF != "" {
```

Comment:
In this way we always test only one option - with storage or not. So always miss one of the cases.

Reply:
This depends on the configuration file. If a storage profile was defined, we use it.